libs/ledger/include/ledger/protocols/main_chain_rpc_protocol.hpp

```diff
@@ -67,8 +68,18 @@ class MainChainProtocol : public service::Protocol
 
   Travelogue TimeTravel(Digest start)
   {
-    auto ret_val = chain_.TimeTravel(std::move(start));
-    return {Copy(ret_val.blocks), ret_val.heaviest_hash};
+    try
+    {
+      auto ret_val = chain_.TimeTravel(std::move(start));
```

Comment:
This allows peers to request a time travel for a missing hash which causes the main chain to throw